python/across_api/burstcube/toorequest.py

```diff
@@ -0,0 +1,437 @@
+
+        # Calculate Ephemeris for the requested dump time at one second time
+        # resolution
+        ephem = BurstCubeEphem(
+            begin=self.trigger_time + self.offset,
+            end=self.trigger_time + self.offset + self.exposure,
+            stepsize=1 * u.s,
+        )
+
+        # Check if the trigger time is in the SAA
+        saa = burstcube_saa_constraint(time=self.trigger_time, ephem=ephem)
+        if saa is True:
+            self.too_info += "Trigger time inside SAA. "
+            self.reject_reason = TOOReason.saa
+            return False
+
+        # Check if the trigger is inside FOV at T0
+        if self.skycoord is not None or self.healpix_loc is not None:
+            fov = BurstCubeFOV(ephem=ephem, time=self.trigger_time)
+            infov = fov.probability_in_fov(
+                skycoord=self.skycoord,
+                error_radius=self.error_radius,
+                healpix_loc=self.healpix_loc,
+                healpix_scheme=self.healpix_scheme,
+            )
+
+            # Flag up if the required probability is not met
+            if infov < self.min_prob:
+                self.too_info += f"Probability inside FOV: {100*infov:.2f}%. Trigger was occulted at T0. "
+                self.reject_reason = TOOReason.earth_occult
+                return False
+            else:
+                self.too_info += f"Probability inside FOV: {100*infov:.2f}%. "
+
+        # Check if any part of the dump time is inside the SAA, warn if so
+        if True in burstcube_saa_constraint(time=ephem.timestamp, ephem=ephem):
+            self.too_info += "Dump time partially inside SAA."
+
+        #  Strip excess whitespace
+        self.too_info = self.too_info.strip()
+        return True
+
+    def post(self) -> bool:
+        """
+        Upload BurstCubeTOO to ACROSS API using POST
+
+        Returns
+        -------
+            Did this work? True | False
+        """
+        # Shouldn't run this unless a created_by is set
+        if self.username is None:
+            raise HTTPException(401, "Username not set.")
+
+        # Validate supplied BurstCubeTOO values against the Schema
+        if not self.validate_post():
+            return False
+
+        # Set the created_on and created_by fields
+        self.created_on = Time.now()
+        self.created_by = self.username
+
+        # Set the ID of this TOO
+        self.id = hashlib.md5(
+            f"{self.trigger_time}{self.exposure}{self.offset}".encode()
+        ).hexdigest()
+
+        # Check if this BurstCubeTOO exists. As the id is just a hash of the
+        # trigger_time, exposure and offset, then repeated requests for values
+        # that match this will be caught.
+        try:
+            response = self.table.get_item(Key={"id": self.id})
+            if response.get("Item"):
+                raise HTTPException(409, "BurstCubeTOO already exists.")
+        except botocore.exceptions.ClientError:
+            pass
```

Comment:
This is a very broad `except:` statement. What errors are you trying to handle?

Reply:
Yeah, not sure honestly. I just removed them, as I don't consistently try to trap exceptions with table read/writes. a474f4a5